atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampUtils.java

```diff
@@ -244,6 +255,60 @@ private static String wrapInQuotes(String tableName) {
+        String message = "Cannot set cached timestamp bound value from {} to {}. The bounds must be increasing!";
+        String formattedMessage = MessageFormatter.arrayFormat(message, new String[]{
+                Long.toString(currentLimit), entryInDb.getTimestampAsString()}).getMessage();
+        logMessage(formattedMessage);
+        throw new IllegalArgumentException(formattedMessage);
+    }
+
+    protected static void throwGettingMultipleRunningTimestampServiceError(UUID id,
+            TimestampBoundStoreEntry entryInDb) {
+        String message = "Error getting the timestamp limit from the DB: the timestamp service ID {} in the DB"
+                + " does not match this service's ID: {}. This may indicate that another timestamp service is"
+                + " running against this cassandra keyspace.";
+        String formattedMessage = MessageFormatter.arrayFormat(message, new String[]{
+                entryInDb.getIdAsString(), id.toString()}).getMessage();
+        logMessage(formattedMessage);
+        throw new MultipleRunningTimestampServiceError(formattedMessage);
+    }
+
+    protected static void throwStoringMultipleRunningTimestampServiceError(long currentLimit, UUID id,
+            TimestampBoundStoreEntry entryInDb, TimestampBoundStoreEntry desiredNewEntry) {
+        String message = "Unable to CAS from {} to {}. Timestamp limit changed underneath us or another timestamp"
+                + " service ID detected. Limit in memory: {}, this service's ID: {}. Limit stored in DB: {},"
+                + " the ID stored in DB: {}. This may indicate that another timestamp service is running against"
+                + " this cassandra keyspace. This is likely caused by multiple copies of a service running without"
+                + " a configured set of leaders or a CLI being run with an embedded timestamp service against"
+                + " an already running service.";
+        String formattedMessage = MessageFormatter.arrayFormat(message, new String[]{
+                Long.toString(currentLimit),
+                desiredNewEntry.getTimestampAsString(),
+                Long.toString(currentLimit),
+                id.toString(),
+                entryInDb.getTimestampAsString(),
+                entryInDb.getIdAsString()})
+                .getMessage();
+        logMessage(formattedMessage);
+        throw new MultipleRunningTimestampServiceError(formattedMessage);
+    }
+
+    protected static void throwUpdateUncheckedException(TimestampBoundStoreEntry entryInDb,
+            TimestampBoundStoreEntry desiredNewEntry, Exception exception) {
+        String message = "[CAS] Error trying to set from {} to {}";
+        String formattedMessage = MessageFormatter.arrayFormat(message, new String[]{
+                entryInDb.getTimestampAsString(), desiredNewEntry.getTimestampAsString()}).getMessage();
+        logMessage(formattedMessage);
+        throw Throwables.throwUncheckedException(exception);
+    }
```

Comment:
Two notes:

- If we go with the earlier suggestions then this should return the exception rather than throwing it.
- These methods should be package visibility (or public, if you intend that); `protected` seems unusual given that this is a final class